src/main/java/pl/byczazagroda/trackexpensesappbackend/service/WalletServiceImpl.java

```diff
@@ -2,12 +2,35 @@
 
 import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
+import org.springframework.validation.annotation.Validated;
+import pl.byczazagroda.trackexpensesappbackend.dto.CreateWalletDTO;
+import pl.byczazagroda.trackexpensesappbackend.dto.WalletDTO;
+import pl.byczazagroda.trackexpensesappbackend.exception.WalletNotSavedException;
+import pl.byczazagroda.trackexpensesappbackend.mapper.WalletModelMapper;
+import pl.byczazagroda.trackexpensesappbackend.model.Wallet;
 import pl.byczazagroda.trackexpensesappbackend.repository.WalletRepository;
 
+import javax.validation.Valid;
+
 @Service
+@Validated
 @RequiredArgsConstructor
 public class WalletServiceImpl implements WalletService {
 
     private final WalletRepository walletRepository;
+    private final WalletModelMapper walletModelMapper;
+
+
+    @Override
+    public WalletDTO createWallet(@Valid CreateWalletDTO createWalletDTO) {
+        String walletName = createWalletDTO.name();
+        Wallet wallet = new Wallet(walletName);
```

Comment:
Think no point to create `String walletName` in this place and better just to use:
```java
Wallet wallet = new Wallet( createWalletDTO.name() );
```

Because one time less is used String constructor.
The correct tradeoff can vary due to circumstances. For example, Sun's Java Virtual Machine has two major modes - client and server. In client mode, minimal compilation and optimization is performed, to reduce startup time. In server mode, extensive compilation and optimization is performed, to maximize performance once the application is running by sacrificing startup time. Other Java just-in-time compilers have used a runtime measurement of the number of times a method has executed combined with the bytecode size of a method as a heuristic to decide when to compile. Still another uses the number of times executed combined with the detection of loops. In general, it is much harder to accurately predict which methods to optimize in short-running applications than in long-running ones.